pyomo/contrib/mindtpy/tests/unit_test.py

```diff
@@ -64,266 +60,227 @@ def test_handle_termination_condition(self):
-            config.fp_main_norm = 'L_infinity'
-            generate_norm_constraint(fp_nlp, solve_data, config)
-            self.assertIsNotNone(fp_nlp.find_component('norm_constraint'))
-
-            # test set_solver_options
-            config.mip_solver = 'gams'
-            config.threads = 1
-            opt = SolverFactory(config.mip_solver)
-            set_solver_options(opt, solve_data, config,
-                               'mip', regularization=False)
-
-            config.mip_solver = 'gurobi'
-            config.mip_regularization_solver = 'gurobi'
-            config.regularization_mip_threads = 1
-            opt = SolverFactory(config.mip_solver)
-            set_solver_options(opt, solve_data, config,
-                               'mip', regularization=True)
-
-            config.nlp_solver = 'gams'
-            config.nlp_solver_args['solver'] = 'ipopt'
-            set_solver_options(opt, solve_data, config,
-                               'nlp', regularization=False)
-
-            config.nlp_solver_args['solver'] = 'ipopth'
-            set_solver_options(opt, solve_data, config,
-                               'nlp', regularization=False)
-
-            config.nlp_solver_args['solver'] = 'conopt'
-            set_solver_options(opt, solve_data, config,
-                               'nlp', regularization=False)
-
-            config.nlp_solver_args['solver'] = 'msnlp'
-            set_solver_options(opt, solve_data, config,
-                               'nlp', regularization=False)
-
-            config.nlp_solver_args['solver'] = 'baron'
-            set_solver_options(opt, solve_data, config,
-                               'nlp', regularization=False)
-
-            # test algorithm_should_terminate
-            solve_data.should_terminate = True
-            solve_data.primal_bound = float('inf')
-            self.assertIs(algorithm_should_terminate(
-                solve_data, config, check_cycling=False), True)
-            self.assertIs(
-                solve_data.results.solver.termination_condition, tc.noSolution)
-
-            solve_data.primal_bound = 100
-            self.assertIs(algorithm_should_terminate(
-                solve_data, config, check_cycling=False), True)
-            self.assertIs(
-                solve_data.results.solver.termination_condition, tc.feasible)
-
-            solve_data.primal_bound_progress = [float('inf'), 5, 4, 3, 2, 1]
-            solve_data.primal_bound_progress_time = [1, 2, 3, 4, 5, 6]
-            solve_data.primal_bound = 1
-            self.assertEqual(get_primal_integral(solve_data, config), 14.5)
-
-            solve_data.dual_bound_progress = [float('-inf'), 1, 2, 3, 4, 5]
-            solve_data.dual_bound_progress_time = [1, 2, 3, 4, 5, 6]
-            solve_data.dual_bound = 5
-            self.assertEqual(get_dual_integral(solve_data, config), 14.1)
+            # MindtPy = solve_data.working_model.MindtPy_utils
```

Comment:
Are these commented out because they fail or do they need to be adapted to the new code?

Reply:
The unit test are used for functions. Now the functions becomes methods in the algorithm base class. So it doesn't work now.